### PR TITLE
Tabs refactor

### PR DESCRIPTION
**Pull request by JJag**
Move tabs from index.html to scala.js components

### CODE REVIEW

File: frontend/src/main/scala/akkaviz/frontend/components/TabMenu.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import scalatags.JsDom.all._
+
+class TabMenu(elemId: String, initTabs: Tab*) extends Component {
+
+  private[this] val navbar = ul(cls := "nav nav-tabs")
+  private[this] val tabContent = div(cls := "tab-content")
+  private[this] val rendered: Element = div(id := elemId, cls := "menu tab-container", navbar, tabContent).render
+
+  private[this] var tabs: Seq[Tab] = Seq()
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/TabMenu.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2349669535/issues/source?bid=3121282&fileBranchId=3217398#l14)

File: frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala [hunk 2]
```diff
@@ -203,30 +205,27 @@ object FrontendApp extends JSApp with Persistence with PrettyJson with Manipulat
...
     repo.newActors.foreach {
       newActors =>
-        hierarchyView.insert(newActors)
+        hierarchyPanel.insert(newActors)
     }
 
+    setupComponents()
+  }
+
+  def setupComponents(): Unit = {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Use active names for operations with side effects](https://www.codacy.com/app/lustefaniak/akka-viz/file/2349669505/issues/source?bid=3121282&fileBranchId=3217398#l214)

File: frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala [hunk 3]
```diff
@@ -203,30 +205,27 @@ object FrontendApp extends JSApp with Persistence with PrettyJson with Manipulat
...
-    messageFilter.attach(document.getElementById("messagefiltering"))
-    messagesPanel.attach(document.getElementById("messagelist"))
-    asksPanel.attach(document.getElementById("asklist"))
-    document.getElementById("receivedelay").appendChild(receiveDelayPanel.render) //FIXME: port to component
-    monitoringOnOff.attach(document.getElementById("onoffsettings"))
-    unconnectedOnOff.attach(document.getElementById("graphsettings"))
-    replTerminal.attach(document.getElementById("repl-tab"))
-    graphView.attach(document.getElementById("graphview"))
-    hierarchyView.attach(document.getElementById("hierarchy-view"))
+    document.getElementById("globalsettings").appendChild(receiveDelayPanel.render) //FIXME: port to component
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Take the required action to fix the issue indicated by this comment.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2349669505/issues/source?bid=3121282&fileBranchId=3217398#l219)

File: frontend/src/main/scala/akkaviz/frontend/components/SettingsTab.scala [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import rx.Var
+
+import scalatags.JsDom.all._
+
+class SettingsTab(
+    monitoringStatus: Var[MonitoringStatus],
+    showUnconnected: Var[Boolean]
+) extends Tab {
+
+  val montiringOnOff = new MonitoringOnOff(monitoringStatus)
```
Comment (pkoryzna) [inline frontend/src/main/scala/akkaviz/frontend/components/SettingsTab.scala]: typo

File: frontend/src/main/scala/akkaviz/frontend/components/MessagesPanel.scala [hunk 5]
```diff
@@ -102,33 +102,38 @@ class MessagesPanel(selectedActors: Var[Set[String]]) extends Component with Pre
...
+  }
+
+  override def name: String = "Messages"
+
+  override def tabId: String = "messages-list"
 
-    parent.appendChild(elem)
+  override def onCreate(): Unit = {
+    tabBody.appendChild(elem)
+    console.log(tabBody)
```
Comment (pkoryzna) [inline frontend/src/main/scala/akkaviz/frontend/components/MessagesPanel.scala]: spam

File: frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala (hunk 2 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Use active names for operations with side effects](https://www.codacy.com/app/lustefaniak/akka-viz/file/2349776081/issues/source?bid=3121282&fileBranchId=3217398#l214)

File: frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala (hunk 3 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/FrontendApp.scala]: Codacy Issue found: [Take the required action to fix the issue indicated by this comment.](https://www.codacy.com/app/lustefaniak/akka-viz/file/2349776081/issues/source?bid=3121282&fileBranchId=3217398#l219)

File: frontend/src/main/scala/akkaviz/frontend/components/TabMenu.scala (hunk 1 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/components/TabMenu.scala]: Codacy Issue found: [Avoid mutable fields](https://www.codacy.com/app/lustefaniak/akka-viz/file/2349776111/issues/source?bid=3121282&fileBranchId=3217398#l14)

**Pull request merged**